File: Module 4 - Data Structure/Set/Challengeproblem/UserID.cpp
/*
Take the IDs as user input and determine:
Total IDs
Unique IDs
Duplicate IDs
Number of duplicates
*/

#include <iostream>
#include <vector>
#include <set>
using namespace std;

int main()
{
    int n, id;
    vector<int> ids;
    set<int> uniqueIds;
    set<int> duplicates;

    cout << "Enter number of IDs: ";
    cin >> n;

    for(int i = 0; i < n; i++)
    {
        cin >> id;

        ids.push_back(id);
        uniqueIds.insert(id);
    }

    for(int id : uniqueIds)
    {
        int count = 0;

        for(int x : ids)
        {
            if(x == id)
                count++;
        }

        if(count > 1)
            duplicates.insert(id);
    }

    cout << "Total IDs: " << ids.size() << endl;
    cout << "Unique IDs: " << uniqueIds.size() << endl;

    cout << "Duplicate IDs: ";
    for(int id : duplicates)
        cout << id << " ";

    cout << "\nNumber of duplicates: " << duplicates.size();

    return 0;
}

File: Module 4 - Data Structure/Set/Challengeproblem/UserID.py
"""
Take the IDs as user input and determine:
Total IDs
Unique IDs
Duplicate IDs
Number of duplicates
"""
set_of_ids = list(map(int, input("Enter the IDs: ").split()))
unique_ids = set(set_of_ids)
duplicate_ids = []

for id in unique_ids:
    if set_of_ids.count(id) > 1:
        duplicate_ids.append(id)

print("Total IDs:", len(set_of_ids))
print("Unique IDs:", len(unique_ids))
print("Duplicate IDs:", duplicate_ids)
print("Number of duplicates:", len(duplicate_ids))
